### ISSUE TITLE
Upgrade react-native-webview to 3.x, 4.x, 5.x

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-mobile#3550]
We're using `react-native-webview` version 2.0.0. In the months since that release, they've made several more with major changes.

In particular @jainkuniya found with `git bisect` that #3518 was introduced by switching from the RN-included WebView to `react-native-webview`. And he found that upgrading all the way to 5.9.1 fixes #3518. (Described on #3536). AFAIK he didn't test versions in between, so it's possible a much earlier version already fixes that issue. Anyway, we should get to the latest.

The latest version of `react-native-webview`, from last week, comes as a pair ([see NPM](https://www.npmjs.com/package/react-native-webview)): 6.3.1 with AndroidX, and 5.12.1 without. Until #3548 , we'll want the non-AndroidX version, so that means the latest 5.x version.

@jainkuniya also described on #3536 finding that upgrading to 5.9.1 wasn't smooth. So let's do this in steps:
* First, to the latest 3.x.
* ~~Then the latest 4.x.~~ -> looking at the version history, 5.0.1 came less than 3 days after 4.0.0, so perhaps skip this.
* Then the latest 5.x.

For each version:
* Read the release notes. Figure out what breaking changes the maintainers say they made.
* Try the upgrade.
* Fix any issues.
* In the commit message, include:
  * a link to the release notes
  * a mention of any breaking changes that we think might matter
  * a description of what you did about them.

A couple of things we know about already:
* @jainkuniya mentioned on #3536 something changed in 5.x with `postMessage`, and linked to https://github.com/react-native-community/react-native-webview/issues/363#issuecomment-467730963 .
* That thread links in turn to another thread: https://github.com/react-native-community/react-native-webview/issues/66#issuecomment-459804483
* In short it looks like `postMessage` has been dropped in favor of an API that's said to be better. We should switch to that new API.

**Comment by gnprice** [comment]
This appears to be a requirement for the RN v0.59 upgrade, #3399 -- versions of `react-native-webview` prior to 5.x say they're only compatible with RN v0.57. (A bit more discussion in #3561 .) Bumping the priority accordingly.

**Comment by gnprice** [comment]
A tactical note: the 5.0.0 release notes say:

"""
*Side note: if you wish to keep compatibility with the old version when you upgrade, you can use the injectedJavascript prop to do that:*

```
const injectedJavascript = `(function() {
  window.postMessage = function(data) {
    window.ReactNativeWebView.postMessage(data);
  };
})()`;
```
"""

So that sounds like the right solution to pursue for now, to unblock the upgrade. Then we can switch to the new, said-to-be-better API at our leisure.
